### DISCUSSION TITLE
Can't figure out how to override region when using Secrets Manager

### DISCUSSION THREAD

**Discussion by hudakinc** [awspring/spring-cloud-aws#1183] [General]
We are using the secrets manager integration to load some app secrets. We have our app running locally against localstack, which is setup to run in region us-west2.

We've tried setting the suggested global region override:

cloud.aws.stack.auto=false
cloud.aws.region.static=us-west-2

However, whatever we've tried, the client is still signing the request with us-east-1 so our secret is never found since it's in the wrong region. The request header looks like this:

'Authorization': 'AWS4-HMAC-SHA256 Credential=fakeAccessKey/20240722/us-east-1/secretsmanager/aws4_request

I'm guessing that localstack is extracting the region from the Auth header as it doesn't seem to appear anywhere else in the request.

I've tried directly setting a region and endpoint override when configuring the secrets manager client but this throws an error since you can only do one or the other.

Does anyone have any idea how to override this successfully?

**Comment by neiljbrown** [comment]
Did you mean or try the alternatively named property spring.cloud.aws.region.static ? (The property name changed from 2.x to 3.x. See https://docs.awspring.io/spring-cloud-aws/docs/3.1.1/reference/html/index.html#migration-from-2-x-to-3-x).

**Reply by hudakinc** [reply]
We are using 2.4.4
